cmd/installer/pkg/install/install.go

```diff
@@ -238,93 +227,111 @@ func NewInstaller(ctx context.Context, cmdline *procfs.Cmdline, mode Mode, opts
 //
 //nolint:gocyclo,cyclop
 func (i *Installer) Install(ctx context.Context, mode Mode) (err error) {
-	errataArm64ZBoot()
+	// pre-flight checks, erratas
+	hostTalosVersion, err := readHostTalosVersion()
+	if err != nil {
+		return err
+	}
 
 	if mode == ModeUpgrade {
-		if err = i.errataNetIfnames(); err != nil {
-			return err
-		}
+		errataArm64ZBoot()
+
+		i.errataNetIfnames(hostTalosVersion)
 	}
 
 	if err = i.runPreflightChecks(mode); err != nil {
 		return err
 	}
 
+	// prepare extensions if legacy machine.install.extensions is present
```

Comment:
should we drop this now or 1.9?

Reply:
I'd say we keep it for now... at least as long as it doesn't break anything